### PR TITLE
 feat: transaction confirmation awaiting

### PR DESCRIPTION
**Pull request by Sokel**
* add utils that await block confirmation

* upd PlaceOrder & OpenDeal funcs

* add custom ethclient

### CODE REVIEW

File: blockchain/api.go [hunk 1]
```diff
@@ -141,7 +140,7 @@ func NewAPI(opts ...Option) (API, error) {
 		return nil, err
 	}
 
-	marketApi, err := NewBasicMarket(clientSidechain, market.MarketAddr(), defaults.gasPriceSidechain, defaults.logParsePeriod)
+	marketApi, err := NewBasicMarket(clientSidechain, market.MarketAddr(), defaults.gasPriceSidechain, defaults.logParsePeriod, 4)
```
Comment (antmat) [inline blockchain/api.go]: make it const at least

File: blockchain/gen_receipt_json.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
+// Code generated by github.com/fjl/gencodec. DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (Sokel) [reply blockchain/gen_receipt_json.go]: Its advanced json marshaller with features which the standard json package cannot offer.
Any docs here:
https://godoc.org/github.com/fjl/gencodec

File: blockchain/receipt.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
+// Copyright 2014 The go-ethereum Authors
```
Comment (antmat) [inline blockchain/receipt.go]: what is the difference with the original code?

File: blockchain/util.go [hunk 4]
```diff
@@ -80,22 +78,61 @@ func findLogByTopic(ctx context.Context, client *ethclient.Client, tx *types.Tra
...
 
-			return log, err
+			return logs, err
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+}
+
+func WaitTransactionReceipt(ctx context.Context, client CustomEthereumClient, confirmations int64, logParsePeriod time.Duration, tx *types.Transaction) (*Receipt, error) {
```
Comment (antmat) [inline blockchain/util.go]: it's rather unclear from the name of this method - what is the difference with the waitForTransactionResult

File: blockchain/gen_receipt_json.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (antmat) [reply blockchain/gen_receipt_json.go]: Is it really worth introducing custom generated non-obvious code from nsome strange tool just for checking several fields for nil?

File: blockchain/receipt.go (hunk 3 above)
Comment (Sokel) [reply blockchain/receipt.go]: only on `BlockNumber` field

Comment (antmat) [reply blockchain/receipt.go]: How about smth like 
```
type SokelReciept struct {
 eth.Receipt
 BlockNumber uint64
}
```

Comment (Sokel) [reply blockchain/receipt.go]: Thanks a lot!

File: blockchain/util.go (hunk 4 above)
Comment (Sokel) [reply blockchain/util.go]: `WaitTransactionReceipt` use Custom client and waiting correct Receipt after confirmations
In other, `waitTransactionResult` stay this as deprecated stuff.
Thanks i would plan to remove this.

Comment (Sokel) [reply blockchain/util.go]: diff so strange, code changed only 3 lines

**Comment by Sokel** [comment]
@sshaman1101 @antmat PTAL

**Pull request merged**